use-cases/virgo/data.py

```diff
@@ -61,159 +67,158 @@ def execute(self) -> pd.DataFrame:
         """
-        dataset = self.get_or_load(dataset)
+        # Load a single dataframe from the file
+        dataframe = pd.read_pickle(self.file_paths[idx])
 
-        # Convert data to torch
-        df = dataset.applymap(lambda x: torch.tensor(x))
+        # Convert all data in the DataFrame to torch tensors
+        df = dataframe.map(lambda x: torch.tensor(x))
 
-        # Divide Image dataset in main and aux channels. Note that df
-        # generated in the section Generate Synthetic Dataset will always have
-        # the main channel as its first column
+        # Divide Image dataset in main and aux channels.
         main_channel = list(df.columns)[0]
         aux_channels = list(df.columns)[1:]
 
+        # Ensure that there are at least 3 auxiliary channels
+        if len(aux_channels) < 3:
+            print(f"Item with the index {idx} only has {len(aux_channels)} channels!")
+            return None
+
+        # Extract the main channel and auxiliary channels
         df_aux_all_2d = pd.DataFrame(df[aux_channels])
         df_main_all_2d = pd.DataFrame(df[main_channel])
-        X_train_2d, X_test_2d, y_train_2d, y_test_2d = train_test_split(
-            df_aux_all_2d, df_main_all_2d,
-            test_size=self.validation_proportion, random_state=self.rnd_seed)
-        return (X_train_2d, y_train_2d), (X_test_2d, y_test_2d), None
 
+        # Stack the main and auxiliary channels into 2D tensors
+        signal_data_train_2d = torch.stack(
+            [torch.stack(
+                [df_main_all_2d[main_channel].iloc[i]]) for i in range(df_main_all_2d.shape[0])])
+
+        aux_data_train_2d = torch.stack(
+            [torch.stack(
+                [df_aux_all_2d.iloc[i, 0],
+                    df_aux_all_2d.iloc[i, 1],
+                    df_aux_all_2d.iloc[i, 2]]
+            ) for i in range(df_aux_all_2d.shape[0])]
+        )
 
-class TimeSeriesProcessor(DataProcessor):
-    def __init__(self, name: str | None = None) -> None:
-        super().__init__(name)
-        self.save_parameters(**self.locals2params(locals()))
+        # Concatenate the main and auxiliary channel tensors
+        data_tensor = torch.cat([signal_data_train_2d, aux_data_train_2d], dim=1)
 
-    @monitor_exec
-    def execute(
+        # Normalize the concatenated tensor
+        data_tensor = normalize_(data_tensor)
+
+        return data_tensor
+
+
+class TimeSeriesDatasetSplitter(DataSplitter):
+    def __init__(
         self,
-        train_dataset: Tuple,
-        validation_dataset: Tuple,
-        test_dataset: Any = None
-    ) -> Tuple[torch.Tensor, torch.Tensor, None]:
-        """Pre-process datasets: rearrange and normalize before training.
+        train_proportion: int | float,
+        validation_proportion: int | float = 0.0,
+        test_proportion: int | float = 0.0,
+        rnd_seed: int | None = None,
+        root_folder: str | None = None,
+        name: str | None = None
```

Comment:
In this case you are better off using Typing's "Union" and "Optional" specifiers. This is also what we have done for the rest of the files in the project.